kythe/java/com/google/devtools/kythe/analyzers/java/KytheTreeScanner.java

```diff
@@ -844,11 +853,24 @@ public JavaNode visitTypeApply(JCTypeApply tApply, TreeContext owner) {
       return new JavaNode(pkgNode);
     } else {
       scan(field.getExpression(), ctx);
-      return emitNameUsage(
-          ctx,
-          sym,
-          field.name.contentEquals("class") ? Keyword.CLASS : field.name,
-          imprt != null ? EdgeKind.REF_IMPORTS : EdgeKind.REF);
+
+      Name name = field.name.contentEquals("class") ? Keyword.CLASS : field.name;
+      if (imprt != null) {
+        return emitNameUsage(ctx, sym, name, EdgeKind.REF_IMPORTS);
+      }
+
+      RefType refType = getRefType(owner.getTree(), sym, field.pos);
+      // RefType can only be READ, WRITE, OR READ_WRITE so node will always have a value set by
+      // at least one call to emitNameUsage. We just have to initialize to null to keep the compiler
+      // happy.
+      JavaNode node = null;
+      if (refType == RefType.READ || refType == RefType.READ_WRITE) {
```

Comment:
Make this a switch statement? It's odd that both branches match `READ_WRITE`.

Reply:
I could potentially do a switch. This is just based on the logic from Cody's implementation for Go.